src/v2/port_mapping.in.rs

```diff
@@ -195,24 +252,59 @@ impl FromStr for PortMapping {
 }
 
 #[test]
-fn port_mapping_should_have_a_string_representation() {
+fn port_mapping_can_be_formatted_as_a_string() {
```

Comment:
Could we maybe considering keeping the existing test case (the one with the round-trip string conversions in both directions) intact, and:

1. Add a new `127.0.0.1:80:80/udp` test case to the round-trip conversions?
2. Add a _non-round-trip_ test case just for things like `127.0.0.1:80:80/tcp`, which should format as `127.0.0.1:80:80`, as per the other suggestions?

I really like the round-trip tests here, because if we maintain two separate lists, somebody will forget to update one direction sooner or later. So the idea is (1) round-trip tests for everything that round-trips, and (2) one-way parsing test cases for syntax we'll never output. What do you think?

Reply:
Yep I like that approach, will do that now.